benchexec/tablegenerator/__init__.py

```diff
@@ -1121,30 +1122,43 @@ def __str__(self):
 
     @classmethod
     def from_list(cls, values):
-        if any([math.isnan(v) for v in values if v is not None]):
+        if any(math.isnan(v) for v in values if v is not None):
             return StatValue(nan, nan, nan, nan, nan, nan)
 
         values = sorted(v for v in values if v is not None)
         if not values:
             return StatValue(0)
 
         values_len = len(values)
-        values_sum = sum(values)
 
-        mean = values_sum / values_len
+        if inf in values and -inf in values:
+            values_sum = nan
+            mean = nan
+            stdev = nan
+        elif inf in values:
+            values_sum = inf
+            mean = inf
+            stdev = inf
+        elif -inf in values:
+            values_sum = -inf
+            mean = -inf
+            stdev = inf
+        else:
```

Comment:
Don't the calculations in the else branch automatically calculate the correct results?

Reply:
Unfortunately not, there's an invalid operation for the type Decimal on `-inf - -inf`, as well as any operation on `nan`.
As an alternative, we could first cast all Decimal values to floats and only use the else-branch,
but I expected that we use Decimals to get a higher precision than floats.